src/core/graph.js

```diff
@@ -134,6 +211,18 @@ export class Graph {
     return this;
   }
 
+  /**
+   * Remove a node from the graph.
+   *
+   * If the node does not exist in the graph, no action is taken and no error
+   * is thrown. (This operation is idempotent.)
+   *
+   * If the node is incident to any edges, those edges must be removed
+   * before removing the node. Attempting to remove a node that is incident
+   * to some edges will throw an error.
+   *
+   * Returns `this` for chaining.
+   */
   removeNode(a: NodeAddressT): this {
```

Comment:
Never seen `removeNode(a: NodeAddressT): this {` syntax before. Guessing it's syntax from unit tests?

Reply:
These are [Flow] types: the argument has to be a `NodeAddressT`, and the
return type is guaranteed to be the `this` object. (The latter is a bit
stronger than just declaring a return type of `Graph`: if we declare a
class `Subgraph` that extends `Graph`, then we’ll know that `Subgraph`’s
`removeNode` method also returns a `Subgraph` and not just a `Graph`.)

It’s not related to unit tests, no.

[Flow]: https://flow.org/